### ISSUE TITLE
ShellCheck mistakenly checks for echo flags inside the string

### ISSUE THREAD

**Issue by burianvlastimil** [koalaman/shellcheck#1399]
#### For bugs
- Rule Id: SC2039
- My shellcheck version: 0.4.6
- [x] I read the issue's wiki page: https://github.com/koalaman/shellcheck/wiki/SC2039
- [x] I tried on shellcheck.net and verified that this is still a problem

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
echo '-k "HotKey": Optional switch requiring one argument.'
```

#### Here's what shellcheck currently says:

> In POSIX sh, echo flags are undefined.

#### Here's what I wanted or expected to see:

Nothing. There is no error in there. ShellCheck mistakenly checks for echo flags inside the string.

**Comment by koalaman** [comment]
Since they are interpreted by the command, there is no difference between a flag and a string. `echo -n` and `echo '-n'` are the same thing.

No implementation of `echo` that I'm aware of will do anything other than write this giant flag/string verbatim, but I'd still argue in favor of `printf`
